frontend/packages/data-portal/app/components/Download/FileFormatDropdown.tsx

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from 'react'
+
+import { I18n } from 'app/components/I18n'
+import { Select, SelectOption } from 'app/components/Select'
+import { useDownloadModalQueryParamState } from 'app/hooks/useDownloadModalQueryParamState'
+import { useI18n } from 'app/hooks/useI18n'
+import { I18nKeys } from 'app/types/i18n'
+
+const FILE_FORMAT_LABEL_I18N: Record<string, I18nKeys> = {
+  '.mrc': 'fileMrc',
+  '.zarr': 'fileOmeZarr',
+  '.ndjson': 'fileNdJson',
+}
+
+const FILE_FORMAT_TOOLTIP_I18N: Record<string, I18nKeys> = {
+  '.mrc': 'fileFormatTooltipMrc',
+  '.zarr': 'fileFormatTooltipZarr',
+  '.ndjson': 'fileFormatTooltipNdJson',
+}
+
+const FILE_FORMAT_ORDER = ['.mrc', '.zarr', '.ndjson']
+
+export function FileFormatDropdown({ fileFormats }: { fileFormats: string[] }) {
```

Comment:
oh wow this is fancier than I expected! could you specify that fileFormats can only contain the literals `.mrc`, `.zarr`, and `.ndjson` tho?